kv/db_test.go

```diff
@@ -222,8 +222,7 @@ func TestKVDBTransaction(t *testing.T) {
 			t.Errorf("expected value %q; got %q", value, gr.ValueBytes())
 		}
 		return nil
-	})
-	if err != nil {
+	}); err != nil {
```

Comment:
I'm not a fan of this pattern where the if expression is multiple lines long due to an embedded closure. I feel like it makes reading the code much more difficult.